assignments/prototypes.js

```diff
@@ -32,79 +49,173 @@
-  console.log(swordsman.healthPoints); // 15
-  console.log(mage.name); // Bruce
-  console.log(swordsman.team); // The Round Table
-  console.log(mage.weapons); // Staff of Shamalama
-  console.log(archer.language); // Elvish
-  console.log(archer.greet()); // Lilith offers a greeting in Elvish.
-  console.log(mage.takeDamage()); // Bruce took damage.
-  console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+const mage = new Humanoid({
+  createdAt: new Date(),
+  dimensions: {
+    length: 2,
+    width: 1,
+    height: 1
+  },
+  healthPoints: 5,
+  name: 'Bruce',
+  team: 'Mage Guild',
+  weapons: ['Staff of Shamalama'],
+  language: 'Common Tongue'
+});
+
+const swordsman = new Humanoid({
+  createdAt: new Date(),
+  dimensions: {
+    length: 2,
+    width: 2,
+    height: 2
+  },
+  healthPoints: 15,
+  name: 'Sir Mustachio',
+  team: 'The Round Table',
+  weapons: ['Giant Sword', 'Shield'],
+  language: 'Common Tongue'
+});
+
+const archer = new Humanoid({
+  createdAt: new Date(),
+  dimensions: {
+    length: 1,
+    width: 2,
+    height: 4
+  },
+  healthPoints: 10,
+  name: 'Lilith',
+  team: 'Forest Kingdom',
+  weapons: ['Bow', 'Dagger'],
+  language: 'Elvish'
+});
+
+console.log(mage.createdAt); // Today's date
+console.log(archer.dimensions); // { length: 1, width: 2, height: 4 }
+console.log(swordsman.healthPoints); // 15
+console.log(mage.name); // Bruce
+console.log(swordsman.team); // The Round Table
+console.log(mage.weapons); // Staff of Shamalama
+console.log(archer.language); // Elvish
+console.log(archer.greet()); // Lilith offers a greeting in Elvish.
+console.log(mage.takeDamage()); // Bruce took damage.
+console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
+
+// Stretch task:
+// * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.
+// * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
+// * Create two new objects, one a villain and one a hero and fight it out with methods!
+function Hero(hAttrs) {
+  Humanoid.call(this, hAttrs);
+  this.strength = hAttrs.strength;
+}
+
+Hero.prototype = Object.create(Humanoid.prototype);
+
+Hero.prototype.reduceHealthBy = function(severity) {
+  return this.healthPoints - severity > 0
+    ? (this.healthPoints -= severity)
+    : (() => {
+        this.healthPoints = 0;
+        return this.destroy();
+      })();
+};
+
+function Villain(vAttrs) {
+  Humanoid.call(this, vAttrs);
+  this.strength = vAttrs.strength;
+}
+
+Villain.prototype = Object.create(Humanoid.prototype);
+
+Villain.prototype.reduceHealthBy = function(severity) {
+  return this.healthPoints - severity * 2 > 0
+    ? (this.healthPoints -= severity * 2)
+    : (() => {
+        this.healthPoints = 0;
+        return this.destroy();
+      })();
+};
```

Comment:
You could make a reduceHealthBy function and assign this to the prototype of both the Hero and villain, then just pass the relevant person into the function when it's called. This would save repeating the function twice.